### DISCUSSION TITLE
疑问：android的ndk-build中LOCAL_LDLIBS，应该如何用xmake表达

### DISCUSSION THREAD

**Discussion by BanKnight** [xmake-io/xmake#2098] [Q&A]
### 描述讨论详情
例如：
LOCAL_LDLIBS    := -llog  -lm -lz   \
        libs/libxxx.a \ 
        libs/libtest.so \

这种使用xmake的什么表达呢？是add_links吗？
我尝试了很多做法，始终会报出很奇怪的报错：
“requires unsupported dynamic reloc R_ARM_REL32; recompile with -fPIC”，
而强行通过add_cflags("-fPIC")后，问题依旧。

请问如何使用呢？

**Comment by waruqi** [comment]
add_links/add_linkdirs 就可以了，-fPIC 跟 add_links 没啥关系。你的代码，以及 link 的 一些 静态 .a 库里面，没有带 -fPIC 编译，需要把这些 .a 库都开 -fPIC 重编下才行

**Comment by BanKnight** [comment]
好的，多谢。
加上   set_policy("check.auto_ignore_flags", false)，重新clean再编就行了。
